qualtran/bloqs/factoring/rsa/rsa_mod_exp.py

```diff
@@ -104,7 +110,9 @@ def make_for_shor(cls, big_n: 'SymbolicInt', g: Optional['SymbolicInt'] = None):
                 g = sympy.symbols('g')
             else:
                 while True:
-                    g = random.randint(2, int(big_n))
+                    if rs is None:
+                        rs = np.random.RandomState()
```

Comment:
sorry again, but these sorts of things can be a _huge_ headache if they slip through:

the random state must be seeded/initialized outside the loop or untold bad things will happen; including repeatedly trying the same number millions of times depending on how the seeding works